src/cv/localization.cpp

```diff
@@ -106,6 +106,97 @@ GPSCoord ECEFLocalization::localize(const ImageTelemetry& telemetry, const Bbox&
     return targetCoord;
 }
 
+
 GPSCoord GSDLocalization::localize(const ImageTelemetry& telemetry, const Bbox& targetBbox) {
-    return GPSCoord();
-}
\ No newline at end of file
+    GPSCoord gps;
+
+    // Ground Sample Distance (mm/pixel), 1.0~2.5cm per px is ideal aka 10mm~25mm ppx
+    double GSD = (SENSOR_WIDTH * (telemetry.altitude)) / (FOCAL_LENGTH_MM * IMG_WIDTH_PX);
+
+    // Midpoints of the image
+    double img_mid_x = IMG_WIDTH_PX / 2;
+    double img_mid_y = IMG_HEIGHT_PX / 2;
+
+    //midpoints of bounding box around the target
+    double target_x = (targetBbox.x1 + targetBbox.x2)/2;
+    double target_y = (targetBbox.y1 + targetBbox.y2)/2;
+
+    // calculations of bearing
+    // L = (distance(middle, bbox))*GSD  
+    double length = (sqrt(pow((target_x - img_mid_x), 2) + pow((target_y - img_mid_y), 2) * GSD));
+
+    //Translate Image Cordinates to Camera Cordinate Frame (Origin to Center of Image instead of Top Left)
+    double target_camera_cord_x = target_x - (IMG_WIDTH_PX / 2);
+    double target_camera_cord_y = (IMG_HEIGHT_PX / 2) - target_y;
+
+    //Angle of Bearing (Angle from north to target)
+    double thetaB = telemetry.heading + atan(target_camera_cord_x / target_camera_cord_y);   
+
+    //Translate bearing to the 3 quadrant if applicable
+    if (target_camera_cord_x < 0 && target_camera_cord_y < 0){
+        thetaB = 180.0 + thetaB;
+    }
+
+    //Finds the offset of the bbox 
+    double calc_cam_offset_x = target_camera_cord_x * GSD * 0.001; //mm to M
+    double calc_cam_offset_y = target_camera_cord_y * GSD * 0.001; //mm to M
+
+    //Calculates the cordinates using the offset
+    GPSCoord calc_coord = CalcOffset((calc_cam_offset_x), (calc_cam_offset_y), (telemetry.latitude), (telemetry.longitude));
+
+    return calc_coord;
+}
+
+/*
+Takes the in two cordinaates and outputs their distance in meters. 
+
+Parameters:
+- lat1/lon1 (First Cordinate)
+- lat2/lon2 (Second Cordinate)
+
+@returns distance in meters
+
+Reference: http://www.movable-type.co.uk/scripts/latlong.html
+*/
+
+double GSDLocalization::distanceInMetersBetweenCords(const double lat1, const double lon1, const double lat2, const double lon2){

```

Comment:
I really like the inclusion of this function for reasoning about accuracy in terms of feet/meters instead of latitude/longitude.